### DISCUSSION TITLE
SOLUTION TO WORMHOLE STUCK ISSUE

### DISCUSSION THREAD

**Discussion by Sandybeach25** [wormhole-foundation/wormhole#458] [Problems / bugs] [answered]
To my fellow Wormholebridge victims:

I found a solution for the problem:
Coins (USDT in my case) getting vanished during bridging from SOL-chain to ETH-chain.

I discovered that my USDT arrived in my Ethereum-address as (wormhole) (including the brackets). This was only visible when looking at my Eth-address in Etherscan.
Some reply of SOL-devs in Github advised me to try to bridge my USDT back to SOL-chain because probably (…) there was no liquidity for USDT.
To be able to send these (wormhole) back I first needed to make them visible in Metamask.

In Etherscan click on the (wormhole) listing in your coinslist.
You arrive at a detail page.
In the detail page look for the contract address and copy that.
Go to Metamask, paste the contract-address in CUSTOM token.
Use WORM as token symbol.
Decimals is usually already filled in (18)
This way I made my WORM visable, including the right amount.

Now go to Wormholebridge from Solana (V2).
Make sure you are transfering FROM Eth-chain TO SOL-chain (in my case).
There use for source input MANUAL and paste there the wormhole token address (the same you used to add it to Metamask).
Connect your Solana wallet (I used Phantom).
I don’t remember the exact details from here anymore, but what I do remember is that I had to sign many times (about3x) in my Metamask, and maybe 5x in my Phantom, but that might be because all my addresses are in hardware wallets.
Also i had to pay fees 2 time in Metamask, 1 time below $ 10.- and 1 time about $ 38.-, and one time in SOL, but that was very cheap.
Also I remembered having to wait a bit (a few minutes) after every signing, so don’t leave the screen and be patient.
The end of this story is that the (wormhole) tokens arrived back in my Phantom wallet as normal USDT.

Hope you guys can profit a bit from my experience, and from posting it here.
May the force be with you!

PS: with special thanks to Tohta78 who pointed me in the right direction.

**Comment by evan-gray** [comment, marked as answer]
Hi @Sandybeach25. Thank you for reporting this! Three notes
- A fix was just released to the site for the issue you encountered with tokens not appearing in the selector on Ethereum. 
- After redeeming on Ethereum, an "Add to Metamask" button should be displayed. This should avoid the manual effort of having to add the token yourself
- There should soon be fixes related to metadata on Solana and then correct info will be recorded on Ethereum for these wrapped assets from Solana
🙂
